src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php

```diff
@@ -647,6 +647,25 @@ public function create_item( $request ) {
 			}
 		}
 
+		/**
+		 * Filters the type of notification sent upon new user registration.
+		 *
+		 * This filter allows customization of the notification type when a new user is created via REST API.
+		 *
+		 * @since 6.9.0
+		 *
+		 * @param string $notify  Determines who gets notified. Accepts:
+		 *                        - 'admin' (default) or an empty string: Notify only the site administrator.
+		 *                        - 'user': Notify only the new user.
+		 *                        - 'both': Notify both admin and user.
+		 *                        - 'false': Disable notifications entirely.
+		 * @param int    $user_id User ID.
+		 */
+		$notify = apply_filters( 'rest_wp_user_created_notification', 'both', $user_id );
+		if ( $notify && 'false' !== $notify ) {
+			wp_new_user_notification( $user_id, null, $notify );
+		}
```

Comment:
```suggestion
		wp_new_user_notification( $user_id, null, $notify );
```

There is already logic in this method. like this. 
```
	// Accepts only 'user', 'admin' , 'both' or default '' as $notify.
	if ( ! in_array( $notify, array( 'user', 'admin', 'both', '' ), true ) ) {
		return;
	}
```

To disable this, all you would have to do, is retrun a value not in this array, so false (bool) would work.